src/relay/analysis/util.cc

```diff
@@ -448,13 +448,8 @@ bool IsDataDependant(const CallNode* call) {
     return false;
   }
 
-  if (op->name == "reshape") {
-    if (const auto* attrs = call->attrs.as<ReshapeAttrs>()) {
-      if (attrs->newshape) {
-        // If newshape attribute exists, it isn't data dependant.
-        return false;
-      }
-    }
+  if (op->name == "dyn.reshape") {
+    return true;
```

Comment:
We can remove line 451-452 as the shape function of `dyn.reshape` is registered to be data dependent.